js/src/annotations/osd-svg-overlay.js

```diff
@@ -0,0 +1,544 @@
+      var transform = 'translate(0px,0px)';
+      this.canvas.style.WebkitTransform = transform;
+      this.canvas.style.msTransform = transform;
+      this.canvas.style.transform = transform;
+      this.canvas.style.marginLeft = pointZero.x + "px";
+      this.canvas.style.marginTop = pointZero.y + "px";
+      if (this.paperScope && this.paperScope.view) {
+        this.paperScope.view.viewSize = new this.paperScope.Size(this.canvas.width, this.canvas.height);
+        this.paperScope.view.zoom = this.viewer.viewport.getZoom(true) / this.initialZoom;
+        this.paperScope.view.center = new this.paperScope.Size(this.paperScope.view.bounds.width / 2, this.paperScope.view.bounds.height / 2);
+        this.paperScope.view.update(true);
+        // Fit pins to the current zoom level.
+        var items = this.paperScope.project.getItems({
+          name: /^pin_/
+        });
+        for (var i = 0; i < items.length; i++) {
+          items[i].scale(this.currentPinSize / this.paperScope.view.zoom);
+        }
+        this.currentPinSize = this.paperScope.view.zoom;
+      }
+    },
+
+    hover: function(event) {
+      if (this.currentTool && event.item && event.item._name.toString().indexOf(this.currentTool.idPrefix) != -1) {
+        this.removeFocus();
+        event.item.selected = true;
+        this.hoveredPath = event.item;
+      }
+    },
+
+    removeFocus: function() {
+      if (this.hoveredPath) {
+        this.hoveredPath.selected = false;
+        this.hoveredPath = null;
+      }
+    },
+
+    restoreEditedShapes: function() {
+      this.editedPaths = [];
+      this.removeFocus();
+    },
+
+    // replaces paper.js objects with the required properties only.
+    replaceShape: function(shape, annotation) {
+      var cloned = new this.paperScope.Path({
+        segments: shape.segments,
+        name: shape.name
+      });
+      cloned.strokeColor = shape.strokeColor;
+      if (shape.fillColor) {
+        cloned.fillColor = shape.fillColor;
+        if (shape.fillColor.alpha) {
+          cloned.fillColor.alpha = shape.fillColor.alpha;
+        }
+      }
+      cloned.closed = shape.closed;
+      cloned.data.rotation = shape.data.rotation;
+      cloned.data.annotation = annotation;
+      if (cloned.name.toString().indexOf('pin_') != -1) { // pin shapes with fixed size.
+        cloned.scale(1 / this.paperScope.view.zoom);
+      }
+      shape.remove();
+      return cloned;
+    },
+
+    parseSVG: function(svg, annotation) {
+      var paperItems = [];
+      var svgParser = new DOMParser();
+      var svgDOM = svgParser.parseFromString(svg, "text/xml");
+      if (svgDOM.documentElement.nodeName == "parsererror") {
+        return; // if svg is not valid XML structure - backward compatibility.
+      }
+      var svgTag = this.paperScope.project.importSVG(svg);
+      // removes SVG tag which is the root object of comment SVG segment.
+      var body = svgTag.removeChildren()[0];
```

Comment:
Every time I try to create a new annotation (say, rectangle) and save it, svgTag is null